integration_test/tests/odbc_2_tests.rs

```diff
@@ -0,0 +1,318 @@
+        let (conn_handle, _, _, _) = connect(env_handle);
+        let mut stmt: Handle = null_mut();
+
+        let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+
+            // check that when requesting all types, both odbc 2 and 3 timestamp types are returned
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::UNKNOWN_TYPE)
+            );
+            for datatype in EXPECTED_DATATYPES {
+                assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt as HStmt));
+                assert_eq!(
+                    SqlReturn::SUCCESS,
+                    SQLGetData(
+                        stmt as HStmt,
+                        2,
+                        CDataType::SLong,
+                        output_buffer as Pointer,
+                        (BUFFER_LENGTH * std::mem::size_of::<u16>() as i16) as Len,
+                        null_mut()
+                    )
+                );
+                assert_eq!(*(output_buffer as *mut i16), datatype.0);
+            }
+
+            assert_eq!(SqlReturn::NO_DATA, SQLFetch(stmt as HStmt));
+
+            // test that SQLGetTypeInfo works properly with odbc 2.x date type
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::EXT_TIMESTAMP)
+            );
+            assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt as HStmt));
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetData(
+                    stmt as HStmt,
+                    2,
+                    CDataType::SLong,
+                    output_buffer as Pointer,
+                    (BUFFER_LENGTH * std::mem::size_of::<u16>() as i16) as Len,
+                    null_mut()
+                )
+            );
+            assert_eq!(*(output_buffer as *mut i16), 11);
+
+            // test that SQLGetTypeInfo returns an error for odbc 3.x date type
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLGetTypeInfo(stmt as HStmt, SqlDataType::TIMESTAMP)
+            );
+        }
+    }
+
+    #[test]
+    fn test_data_retrieval() {
+        let env_handle = setup();
+        connect(env_handle);
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = connect(env_handle);
+        let mut stmt: Handle = null_mut();
+
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+            assert_eq!(
```

Comment:
Why do you check that you can retrieve all type info again in here if what you are trying to validate is the type mapping in SQLGetData?

Reply:
Just a bad copy/paste of the previous function when I was building off of the statement allocation 😓